demo.html

```diff
@@ -0,0 +1,262 @@
+                        }
+
+                        p1 = project(x, y, evalPts[ix][iy])
+                        p2 = project(x + (x_Ra[1] - x_Ra[0]) / steps, y, evalPts[ix + 1][iy])
+                        p3 = project(x + (x_Ra[1] - x_Ra[0]) / steps, y + (y_Ra[1] - y_Ra[0]) / steps, evalPts[ix + 1][iy + 1])
+                        p4 = project(x, y + (y_Ra[1] - y_Ra[0]) / steps, evalPts[ix][iy + 1])
+
+                        c.beginPath()
+                        var w4 = w / 4
+                        var h4 = h / 4
+                        c.moveTo(w4 * (2 * p1[0] + 1), h4 * (2 * p1[1] + 1))
+                        c.lineTo(w4 * (2 * p2[0] + 1), h4 * (2 * p2[1] + 1))
+                        c.lineTo(w4 * (2 * p3[0] + 1), h4 * (2 * p3[1] + 1))
+                        c.lineTo(w4 * (2 * p4[0] + 1), h4 * (2 * p4[1] + 1))
+                        c.lineTo(w4 * (2 * p1[0] + 1), h4 * (2 * p1[1] + 1))
+                        c.fill()
+
+                        iy++
+                        y += sizes[1]
+
+                    }
+
+                    ix++
+                    x += sizes[0]
+
+                }
+
+            }
+
+            var refresh = () => {
+                // 0.005 represent rotation speed
+                theta += 0.005
+                tanTheta = Math.tan(theta)
+                tanThetaSquared = tanTheta * tanTheta
+                theta %= 6.28318
+
+                var val = Math.sqrt(1 / (tanThetaSquared + 1))
+                var eval = Math.sqrt(-(i_hat[0] * i_hat[0]) + 1)
+                if (tanTheta >= 0) {
+
+                    if (theta > 3.14159) {
+
+                        i_hat[0] = -val
+                        i_hat[1] = -eval
+
+                    } else {
+
+                        i_hat[0] = val
+                        i_hat[1] = eval
+
+                    }
+
+                } else {
+
+                    i_hat[0] = -val
+                    i_hat[1] = eval
+
+                }
+
+                var nTheta = (theta + 1.570795) % 6.28318
+                var tanNTheta = Math.tan(nTheta)
+                var tanNThetaSquared = tanNTheta * tanNTheta
+                var val2 = Math.sqrt(1 / (tanNThetaSquared + 1))
+                var eval2 = Math.sqrt(-(j_hat[0] * j_hat[0]) + 1)
+                if (tanNTheta >= 0) {
+
+                    if (nTheta > 3.14159) {
+
+                        j_hat[0] = -val2
+                        j_hat[1] = -eval2
+
+                    } else {
+
+                        j_hat[0] = val2
+                        j_hat[1] = eval2
+
+                    }
+
+                } else {
+
+                    j_hat[0] = -val2
+                    j_hat[1] = eval2
+
+                }
+
+                c.fillStyle = "white"
+                c.fillRect(0, 0, w, h)
+
+                graph(f, 64)
```

Comment:
```suggestion
                graph(f, 32)
```
Using a 32x32 grid will greatly speed up graphing, especially with the animations and everything. If you want, we could add a slider to change the grid size so that it displays better on slower computers.

Reply:
Yes, good point. I will add a slider in my next commit.

Reply:
However, I do believe 64 squares is a good starting amount since I am easily able to render it with my 1.10 GHZ and 4 GB of RAM laptop.

Reply:
Great!